lib/child_process.js

```diff
@@ -50,7 +50,20 @@ exports.fork = function(modulePath /*, args, options*/) {
 
   args = execArgv.concat([modulePath], args);
 
-  if (!Array.isArray(options.stdio)) {
+  if (typeof options.stdio === 'string') {
+    const stdioStringToArray = (option) => {
+      return [option, option, option, 'ipc'];
+    };
+    switch (options.stdio) {
+      case 'ignore':
+      case 'pipe':
+      case 'inherit':
+        options.stdio = stdioStringToArray(options.stdio);
+        break;
+      default:
+        throw new TypeError(`Incorrect type of stdio option: ${options.stdio}`);
```

Comment:
The fixed string part should be greppable and unique, as you say. Just tested, the stringification seems to just do `[object Object]` for recursive and complex objects. Not useful, but not damaging.

`git grep TypeError lib` is pretty clear: https://gist.github.com/f81b6d3e19ab6595244905575e55e082 

There are many, many lines, and all the errors say WHAT is wrong with the input, but do not re-include the input itself in the message.

```
git grep TypeError.*\` lib       
lib/dns.js:    throw new TypeError(`"port" should be >= 0 and < 65536, got "${port}"`);
lib/internal/url.js:      throw new TypeError('Value of `this` is not a URLSearchParams');
... (all examples of `this`, not of format strings)
```

So at this point, we have one example of a format string inputting an invalid argument into the type error message, and this would be the second, so now I'm convinced. The error should not echo the input back, it should include information that is *not* known to the caller - what is wrong with the input, not the input itself.